src/DocumentStore.zig

```diff
@@ -594,6 +622,23 @@ fn getOrLoadBuildFile(self: *DocumentStore, uri: Uri) ?*BuildFile {
     return gop.value_ptr.*;
 }
 
+pub fn associateBuildFileUri(self: *DocumentStore, handle: *Handle) error{OutOfMemory}!void {
+    handle.impl.lock.lock();
+    defer handle.impl.lock.unlock();
+
+    if (handle.impl.associated_build_file != .pending) return;
+
+    handle.impl.associated_build_file = blk: for (handle.potential_build_file_uris.items, 0..) |build_file_uri, index| {
+        const build_file = self.getOrLoadBuildFile(build_file_uri, handle.uri) orelse break :blk .pending;
+        if (!build_file.hasConfig()) break :blk .pending;
+        if (try self.uriAssociatedWithBuild(build_file, handle.uri)) break :blk .{ .uri = handle.potential_build_file_uris.items[index] };
+    } else .none;
```

Comment:
I'm still to implement the `TODO` but I'm thinking
```zig
pending: struct {
	potential: []const Uri,
    last_checked_index: usize,
};
```
to avoid extra allocations?